src/components/FilmsSearch/FilmsSearch.tsx

```diff
@@ -16,6 +16,14 @@ const genres: Genres[] = [
 	{ id: "10752", text: "Wojenny" },
 ];
 
+const checkGenre = (
+	e: React.ChangeEvent<HTMLSelectElement>,
+	setSelectedGenre: { (genres: string[]): void }
+) => {
+	const options = [e.target.value];
+	setSelectedGenre(options);
+};
```

Comment:
A czemu w onChange wewnątrz komponentu od razu nie wywołujesz tego setSelectedGenre?

Tzn ogólnie możesz to mieć w środku komponentu (poza JSX), tylko co to robi poza opakowaniem wartosci w array? Nazwa również trochę myli, powinno być coś w rodzaju onGenreAdd albo handleAddGenre i to powinno byc przekazane do onChange={handleAddGenre}